utils/python/CIME/case.py

```diff
@@ -128,13 +128,10 @@ def initialize_derived_attributes(self):
         self.tasks_per_numa = int(math.ceil(self.tasks_per_node / 2.0))
         smt_factor = max(1,int(self.get_value("MAX_TASKS_PER_NODE")/self.get_value("PES_PER_NODE")))
 
-        self.cores_per_task = ((self.get_value("MAX_TASKS_PER_NODE")/smt_factor) \
-                               / self.tasks_per_node) * 2
+        self.cores_per_task = self.thread_count / smt_factor
```

Comment:
This flag is not needed for cori - all tests pass without it.  But the new computation in line 131 is still incorrect - ```cores_per_task = 1 if (pestot < PES_PER_NODE) else smt_factor```